### PR TITLE
Proj ordering by stars

### PR DESCRIPTION
**Pull request by mscuthbert**
Ordering projects by stars is becoming a time consuming problem for each new project, so I've added star numbers in comments after each project and tried to give a sense of scope for projects not on github.  (And generally left in the same place anything that really requires a community judgement call).

Some projects were listed far too low (very few too high)

### CODE REVIEW

**Comment by takluyver** [comment]
Thanks - I had noticed one or two which had gained a lot of stars since they were added.

Adding star counts definitely makes it easier to add a project now, but it's more information which will get outdated. So I'm ambivalent about that bit.

I suspect search result counts are too inconsistent to include. E.g. you've put '393,000 search results' by NetworkX. I guess that's from Google? I just tried, and I get 'About 781,000 results'. :-/

One of the projects added recently used download counts from https://pypistats.org/ to place itself, as it wasn't on Github. Obviously that only works for downloadable Python packages, but at least different people can hopefully get the same number on the same day.

**Comment by hugovk** [comment]
As an example, I mentioned the stars in some recent commit messages.

https://github.com/python3statement/python3statement.github.io/commit/d9dee776def7baa93b014d7cd49e2eee16c5bc38 April 9th:

> Add Requests (★ 38,021) before pandas (★ 18,865)

Requests is now at 38,713, pandas at 19,630.

https://github.com/python3statement/python3statement.github.io/commit/1214bfaff7b5f2764f10de92af202dca1d0b7386 April 1st:

> Add Pillow (★ 6,015) before Scipy (★ 5,616)

Pillow is now at 6,220, Scipy at 5,807.

**Comment by mscuthbert** [comment]
I suppose that star counts that get out of date would be an issue, but it seems like a small change is likely before Dec 2020, while the number of wildly misplaced projects was substantial (Angr was one of the most notable; placed near the end with 2,000+ stars)

**Comment by takluyver** [comment]
If you want to take out the counts of search results (see above), I'd be +0.5 on this. I'd like @Carreau to weigh in, though.

**Comment by Carreau** [comment]
Both are fine with me; I think we want a "rough ordering", and stars are mostly a proxy. I'll see if I can fix the conflicts and merge this, we can further iterate later.

Thanks !

**Comment by Carreau** [comment]
I'll resolve the conflict later, everything was messed up; git can't figure out which is which.

**Comment by mscuthbert** [comment]
I think I can get the file fixed enough that a merge should be possible w/o losing any projects.  Will double check.

**Comment by Carreau** [comment]
There are conflicts, I'm going to see if I can help you do this by chunks.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,8 @@
 _site/
 .DS_Store
 *.swo
+.idea/misc.xml
+.idea/modules.xml
+.idea/python3statement.github.io.iml
+.idea/vcs.xml
+.idea/workspace.xml
```
Comment (Carreau) [inline .gitignore]: Side note, usually you want to put those in your machine global gitignore.

Comment (mscuthbert) [reply .gitignore]: Ah!  I had no idea there was such a thing.  Thank you!

**Comment by mscuthbert** [comment]
The current version committed here has every change up to yesterday (I don't think there have been merges since then) -- I'm surprised it's still saying there are conflicts since it has all commits merged in it.

**Comment by Carreau** [comment]
We ca also probably go with somthing like 

```
import re
import requests

import requests_cache

requests_cache.install_cache('demo_cache')

match = re.compile('github.com/([a-zA-Z-]+)/([0-9a-zA-Z-]+)/?')

for line in data.splitlines():
    if not 'github' in line:
        continue

    m = match.findall(line)
    if m: 
        org,repo = m[-1]
        url = f'https://api.github.com/repos/{org}/{repo}'
        info = requests.get(url).json()
    line =     line.split('<!')[0]
    print(line + f"       <!-- url:{url} sg:{info['stargazers_count']} -->")

  ```

That should be close to idempotent, and if we manually add a comment with `<!-- url:>` will pick it up.

I'm out of github request for the next hour... I should have authenticated and push the cache first thing.

**Comment by Carreau** [comment]
> The current version committed here has every change up to yesterday (I don't think there have been merges since then) -- I'm surprised it's still saying there are conflicts since it has all commits merged in it.

I was surprised as well; Try to "resolved conflict in github UIs but it got stuck; so maybe a bug on GH ends ? And because there was conflicts I merged some of the PRs that arrived in the  meantime.

**Comment by Carreau** [comment]
Thoughts about #228 ? It just recompute the star for now, but we can work on auto-reordering as well.

**Comment by mscuthbert** [comment]
Closing in favor of #228   Will add information that's missing in that PR from this one after it's done.

**Pull request closed without merging**